packages/clay-datepicker/src/ClayDatepicker.js

```diff
@@ -0,0 +1,384 @@
+import classnames from 'classnames';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React, {useState, useEffect, useRef} from 'react';
+
+import {useCurrentMonth, useCurrentTime, useWeeks} from './Hooks';
+
+import Button from './Button';
+import DateNavigation from './DateNavigation';
+import Day from './Day';
+import Dropdown from './Dropdown';
+import InputDate from './InputDate';
+import TimePicker from './TimePicker';
+import Week from './Week';
+import Weekday from './Weekday';
+import Weekdays from './Weekdays';
+
+/**
+ * ClayDatepicker component.
+ * @return {React.createElement}
+ */
+function ClayDatepicker({
+	ariaLabel,
+	dateFormat,
+	defaultValue,
+	expanded: initialExpanded,
+	firstDayOfWeek,
+	footerElement,
+	id,
+	initialMonth,
+	inputName,
+	months,
+	onValueChange,
+	placeholder,
+	spritemap,
+	time,
+	timeFormat,
+	useNative,
+	weekdaysShort,
+	years,
+}) {
+	/**
+	 * Indicates the current month rendered on the screen.
+	 * @type {!Date}
+	 */
+	const [currentMonth, setCurrentMonth] = useCurrentMonth(initialMonth);
+
+	/**
+	 * Indicates the time selected by the user.
+	 * @type {!Date}
+	 */
+	const [currentTime, setCurrentTime] = useCurrentTime(timeFormat);
+
+	/**
+	 * The day selected by the user.
+	 * @type {!Date}
+	 */
+	const [daySelected, setDaySelected] = useState(initialMonth);
+
+	/**
+	 * The input value.
+	 * @type {?(Date|undefined)}
+	 */
+	const [inputValue, setInputValue] = useState(undefined);
+
+	/**
+	 * An array of the weeks and days list for the current month
+	 * @type {!Array<Array>}
```

Comment:
Not sure what the limitations of the type syntax here are, but can we specify the type of the nested array?

Reply:
Yes we can, but it will not be very readable.